### ISSUE TITLE
move SE API key to database for ease of testing

### ISSUE THREAD

**Issue by jackdouglas** [topanswers/topanswers#73]
per https://github.com/topanswers/topanswers/issues/71 comments

**Comment by paxcodes** [comment]
I assume you plan it to be in the `one` table with the secret key? Won't it be easier and simpler to have it in a [config file / as an environment variable](https://github.com/topanswers/topanswers/blob/master/config.php)? It'll be easier for other contributors as well when setting up their own instance.

I'm also just talking as someone who is not a DB admin. 😅

**Comment by jackdouglas** [comment]
Yes it might makes sense to put it in there — though it isn't any use without the corresponding secret…

**Comment by paxcodes** [comment]
Right. So devs will have their own API key + secret that they will use but the app will load it as an environment variable / through the config file. They will not be hard-coded anywhere.

**Comment by jackdouglas** [comment]
> The API key can be hard-coded there as the default. But other devs can specify their own API key + secret as an environment variable

looks good to me

> If it is preferred that production secrets are placed in the database instead of it being an env variable, we can make a DB call as the default value of SE_SECRET_KEY.

I'm not sure I really do prefer that — it's just easier for me (being a database person) to dump things in there if I'm not sure where to put them!

So, `one.one_encryption_key` probably should become an environment variable in the same way as the SE_SECRET_KEY. On the other hand something like `one.one_image_hash` is naturally at home inside the database, so we can't get rid of `one` altogether.
